src/navigation/navigation_main.cc

```diff
@@ -844,48 +903,48 @@ int main(int argc, char** argv) {
       n.subscribe("halt_robot", 1, &HaltCallback);
   ros::Subscriber override_sub =
       n.subscribe("nav_override", 1, &OverrideCallback);
-
+  //ros::Subscriber path_sub = n.subscribe("/move_base/TrajectoryPlannerROS/global_plan",1,&PathCallback);
   std_msgs::Header viz_img_header; // empty viz_img_header
   viz_img_header.stamp = ros::Time::now(); // time
   cv_bridge::CvImage viz_img;
   if (params.evaluator_type == "cost_map") {
     viz_img = cv_bridge::CvImage(viz_img_header, sensor_msgs::image_encodings::RGB8, navigation_.GetVisualizationImage());
   }
-  
   RateLoop loop(1.0 / params.dt);
+
+  RateChecker rate_checker("Navigation main loop");
   while (run_ && ros::ok()) {
+    RateChecker::Invocation rc_iteration(&rate_checker);
     visualization::ClearVisualizationMsg(local_viz_msg_);
     visualization::ClearVisualizationMsg(global_viz_msg_);
     ros::spinOnce();
 
     // Run Navigation to get commands
     Vector2f cmd_vel(0, 0);
     float cmd_angle_vel(0);
-    bool nav_succeeded = navigation_.Run(ros::Time::now().toSec(), cmd_vel, cmd_angle_vel);
+    navigation_.Run(ros::Time::now().toSec(), cmd_vel, cmd_angle_vel);
```

Comment:
We should still get the success flag and think harder about what to do when it fails. We can't just not publish visualizations if it fails. Need a better way to handle failures.